src/main/python/systemds/matrix/matrix.py

```diff
@@ -65,13 +65,16 @@ def code_line(self, var_name: str, unnamed_input_vars: Sequence[str],
             code_line = code_line.format(file_name=var_name)
         return code_line
 
-    def compute(self, verbose: bool = False) -> Union[np.array]:
+    def compute(self, verbose: bool = False, lineage: bool = False) -> Union[np.array]:
         if self._is_numpy():
             if verbose:
                 print('[Numpy Array - No Compilation necessary]')
             return self.np_array
         else:
-            return super().compute(verbose)
+            return super().compute(verbose, lineage)
+
+    def getLineageTrace(self) -> str:
+        return super().getLineageTrace()
```

Comment:
I think you do not need the lines, because they will anyway be inherited from `OperationNode`

Reply:
Or you might want to add an if/else for the case that we have a numpy array (which we would not pass to systemds if `.compute()` is called directly on this `Matrix` object)